src/Operators/finitedifference.jl

```diff
@@ -1787,7 +1989,182 @@ Base.@propagate_inbounds function stencil_right_boundary(
     return Geometry.Contravariant3Vector(zero(eltype(eltype(A_field))))
 end
 
+"""
+    U = TVDSlopeLimitedFlux(;boundaries)
+    U.(𝒜, Φ, 𝓊)
+𝒜, following the notation of Durran (Numerical Methods for Fluid
+Dynamics, 2ⁿᵈ ed.) is the antidiffusive flux given by
+𝒜 = ℱʰ - ℱˡ
+where h and l superscripts represent the high and lower order (monotone) 
+fluxes respectively. The effect of the TVD limiters is then to 
+adjust the flux 
+F_{j+1/2} = F^{l}_{j+1/2} + C_{j+1/2}(F^{h}_{j+1/2} - F^{l}_{j+1/2})
+where C_{j+1/2} is the multiplicative limiter which is a function of 
+the ratio of the slope of the solution across a cell interface.
+C=1 recovers the high order flux.
+C=0 recovers the low order flux. 
+
+Supported limiter types are 
+- RZeroLimiter (returns low order flux)
+- RHalfLimiter (flux multiplier == 1/2)
+- RMaxLimiter (returns high order flux)
+- MinModLimiter
+- KorenLimiter
+- SuperbeeLimiter
+- MonotonizedCentralLimiter
+"""
+abstract type AbstractTVDSlopeLimiter end
+struct RZeroLimiter <: AbstractTVDSlopeLimiter end
+struct RHalfLimiter <: AbstractTVDSlopeLimiter end
+struct RMaxLimiter <: AbstractTVDSlopeLimiter end
+struct MinModLimiter <: AbstractTVDSlopeLimiter end
+struct KorenLimiter <: AbstractTVDSlopeLimiter end
+struct SuperbeeLimiter <: AbstractTVDSlopeLimiter end
+struct MonotonizedCentralLimiter <: AbstractTVDSlopeLimiter end
+
+@inline function compute_limiter_coeff(r, ::RZeroLimiter)
+    return zero(eltype(r))
+end
+
+@inline function compute_limiter_coeff(r, ::RHalfLimiter)
+    return one(eltype(r)) * 1 / 2
+end
+
+@inline function compute_limiter_coeff(r, ::RMaxLimiter)
+    return one(eltype(r))
+end
+
+@inline function compute_limiter_coeff(r, ::MinModLimiter)
+    return max(zero(eltype(r)), min(one(eltype(r)), r))
+end
+
+@inline function compute_limiter_coeff(r, ::KorenLimiter)
+    return max(zero(eltype(r)), min(2r, min(1 / 3 + 2r / 3, 2)))
+end
+
+@inline function compute_limiter_coeff(r, ::SuperbeeLimiter)
+    return max(zero(eltype(r)), min(one(eltype(r)), r), min(2, r))
+end
+
+@inline function compute_limiter_coeff(r, ::MonotonizedCentralLimiter)
+    return max(zero(eltype(r)), min(2r, (1 + r) / 2, 2))
+end
+
+struct TVDSlopeLimitedFlux{BCS} <: AdvectionOperator
+    bcs::BCS
+end
+
+TVDSlopeLimitedFlux(; method, kwargs...) =
+    TVDSlopeLimitedFlux((; method, kwargs...))
+
+return_eltype(::TVDSlopeLimitedFlux, A, Φ, 𝓊) =
+    Geometry.Contravariant3Vector{eltype(eltype(A))}
+
+return_space(
+    ::TVDSlopeLimitedFlux,
+    A_space::AllFaceFiniteDifferenceSpace,
+    Φ_space::AllCenterFiniteDifferenceSpace,
+    𝓊_space::AllFaceFiniteDifferenceSpace,
+) = A_space
+
+function tvd_limited_flux(Aⱼ₋₁₂, Aⱼ₊₁₂, ϕⱼ₋₁, ϕⱼ, ϕⱼ₊₁, ϕⱼ₊₂, rⱼ₊₁₂, method)
```

Comment:
This function should also probably be compatible with `RecursiveApply`.

Reply:
I discussed this with @dennisYatunin and @akshaysridhar. There are some ergonomic issues with using conditions and RecursiveApply for algorithms. RecursiveApply is nice in that it allows us to operate on NamedTuples, but conditions on NamedTuples becomes more complicated (and entangled with the algorithm). Furthermore, from a parallelization point of view, RecursiveApply couples variables (e.g., a given thread must read multiple variables, even if the algorithm can be performed in parallel across variables).

I think relaxing the use of RecursiveApply here is okay, until we determine if we really need support for this.